packages/interaction-kit/src/commands/options/option.ts

```diff
@@ -88,36 +78,60 @@ export default class Option
 	}
 
 	serialize(): APIApplicationCommandOption {
-		// TypeScript and discord-api-types don't play well with our usage
-		// of generic fields for the `type` field. We need to cast instead
 		// eslint-disable-next-line @typescript-eslint/consistent-type-assertions
-		const payload = {
+		return {
 			type: this.type,
 			name: this.name,
 			description: this.description,
-			required: this.required,
-		} as APIApplicationCommandBasicOption;
-
-		// TODO: Handle Options (do we want to?)
-
-		return payload;
+		} as APIApplicationCommandOption;
 	}
 
 	equals(schema: APIApplicationCommandOption): boolean {
 		if (
 			this.type !== schema.type ||
 			this.name !== schema.name ||
-			this.description !== schema.description ||
-			this.required !== (schema.required ?? false)
+			this.description !== schema.description
 		) {
 			return false;
 		}
 
-		// TODO: Nothing uses options yet, but eventually verify they're correct somehow
-		// if ((this.options?.length ?? 0) !== (schema.options?.length ?? 0)) {
-		// 	return false;
-		// }
-
 		return true;
 	}
 }
+
+interface BasicOptionArgs extends OptionArgs {
+	required: boolean | undefined;
+}
+
+export type BaseBasicOptionArgs = Omit<BasicOptionArgs, "type">;
+
+export class BasicOption<
+	T extends APIApplicationCommandBasicOption = APIApplicationCommandBasicOption
+> extends Option {
+	public readonly required;
+
+	constructor({ name, description, type, required }: BasicOptionArgs) {
+		super({
+			type,
+			name,
+			description,
+		});
+		this.required = required;
+	}
+
+	serialize(): T {
+		if (this.required === undefined) {
```

Comment:
This feels...redundant... All options have required as `required?: boolean | undefined` in -types (at least if you use the right types, which are in the rest body)

Reply:
Roger that, left a note